samcli/commands/local/lib/sam_function_provider.py

```diff
@@ -123,11 +124,18 @@ def _convert_sam_function_resource(name, resource_properties, layers):
 
         LOG.debug("Found Serverless function with name='%s' and CodeUri='%s'", name, codeuri)
 
+        timeout = resource_properties.get("Timeout")
+        if isinstance(timeout, str):
+            try:
+                timeout = ast.literal_eval(timeout)
+            except SyntaxError:
+                raise ValueError('Invalid Number for Timeout: {}'.format(timeout))
```

Comment:
This will raise a ValueError up the chain and wont be caught. This will show up as a command failure with a stack track. We should raise a custom error/exception here to avoid that and provide a better customer experience.

Reply:
Is there an existing exception used for this? Is there an existing convention for how this is handled in the codeline?

Reply:
@setrofim You can follow what we are currently doing for Invalid Layer ARNs.

We throw [there](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/local/lib/sam_function_provider.py#L247) and the exception is defined [here](https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/local/cli_common/user_exceptions.py#L57)

There is currently an exception for `InvalidSamTemplateException`. This might be sufficient here with a good error message (what you currently have is fine). You can define a more specific to improve code readability but either works.

When defining exceptions, make sure the inherit from `UserException`. This will make sure the correct information is added to the exception so Click will work as expected (no stack trace and printing the error message to console). Only other this is to just update your current unit test to expect the new exception instead of ValueError.

Reply:
Sorry for the delayed response. I have now made the change -- I stuck with `InvalidSamTemplateException` as that see to cover this case.